src/runtime/posix_allocator.cpp

```diff
@@ -8,16 +8,36 @@ extern "C" {
 extern void *malloc(size_t);
 extern void free(void *);
 
-WEAK void *halide_default_malloc(void *user_context, size_t x) {
+WEAK void *halide_default_malloc(void *user_context, size_t user_size) {
     // Allocate enough space for aligning the pointer we return.
     const size_t alignment = halide_malloc_alignment();
-    void *orig = malloc(x + alignment);
+
+    // All bets are off if alignment is smaller than a pointer.
+    halide_debug_assert(user_context, alignment >= sizeof(void *));
+
+    // Always round allocations up to alignment size,
+    // so that all allocators follow the behavior of aligned_alloc() and
+    // return aligned pointer *and* aligned length.
+    const size_t aligned_size = align_up(user_size, alignment);
+
+    // We can save a bit of space by special-casing allocations < alignment
+    // in size, which we can always fit entirely into 2*alignment.
+    const size_t requested_size = (aligned_size <= alignment) ?
```

Comment:
Isn't this just: requested_size = aligned_size + alignment? I doubt the -1 is going to do anything.

Reply:
Done